src/game_battler.cpp

```diff
@@ -524,8 +562,11 @@ int Game_Battler::GetSpi() const {
 	int n = min(max(base_spi, 1), 999);
 
 	for (int16_t i : GetInflictedStates()) {
-		if(Data::states[i - 1].affect_spirit) {
-			n = AffectParameter(Data::states[i - 1].affect_type, base_spi);
+		// States are guaranteed to be valid
+		const RPG::State& state = *ReaderUtil::GetElement(Data::states, i);
+
```

Comment:
Stray newline or missing in others.

Try to keep `GetAtk`, `GetDef()`, `GetSpi()` and `GetAgi()` the same.